tests/tensorflow2/test_keras.py

```diff
@@ -588,6 +588,33 @@ def test_layer_names(out_dir, tf_eager_mode):
         assert re.match(pattern=pattern, string=tname) is not None
 
 
+@pytest.mark.slow
+def test_regex_filtering_for_default_collections(out_dir):
+    hook = smd.KerasHook(
+        out_dir,
+        save_config=SaveConfig(save_interval=9),
+        include_collections=[CollectionKeys.LAYERS, CollectionKeys.GRADIENTS],
+    )
+    hook.get_collection("layers").include("^dense")
+    hook.get_collection("gradients").include("gradients/dense")
+    # hook.get_collection(CollectionKeys.LAYERS).include("dense")
```

Comment:
Why use
```
hook.get_collection("layers").include("^dense")
hook.get_collection("gradients").include("gradients/dense")
```
instead of
```
hook.get_collection(CollectionKeys.LAYERS).include("^dense")
hook.get_collection(CollectionKeys.GRADIENTS).include("gradients/dense")
```

If this is the expected way for users to specify regex for each collection, then we should be consistent since the hook's `include_collections` parameter seems to accept enum instead of string. Let's have the `get_collection` function accept enum as well.